app/controllers/links_controller.rb

```diff
@@ -7,6 +7,17 @@ def show
     redirect_to(url, allow_other_host: true)
   end
 
+  def fetch
```

Comment:
This seems very specific. 
Did you want to get two urls (forum and UpgradeTo) in the same response?